cmd/geth/main.go

```diff
@@ -223,6 +230,40 @@ func makeCLIApp() (app *cli.App) {
 
 		runtime.GOMAXPROCS(runtime.NumCPU())
 
+		// log-rotation options
```

Comment:
Maybe we can shove this logic off to its own file `cmd/geth/log_context.go`? I've also got a bunch of this kind of stuff from the display logs too and it feels yucky to clutter `main.go` with all this logic.